### DISCUSSION TITLE
Need possibiliy to add (complex) logical OR dependencies.

### DISCUSSION THREAD

**Discussion by Neumann-A** [microsoft/vcpkg#17784] [Ideas]
This is needed for openblas with lapack, clapack and lapack with blas. 

So that I can write dependencies like openblas,lapack or openblas[lapack] or lapack[blas] or openblas,clapack. The preferred dependency is the first one in the list. If that one cannot be build try the second one and so on.

**Comment by cenit** [comment]
Isn’t it better to write dependencies like these using metapackages, which are resolved (each one) to specific implementations using features?

**Comment by Neumann-A** [comment]
@cenit hmm yeah that is a possibility make something a default feature and the rest optional features. (which could even be mutually exclusive). The downstream ports than depend on meta[core] only. Could you not use the same for your SSL fix in #6730. 
I also figured out that openblas[lapack] is a bad example because it uses the lapack-reference implementation and that should not be added as a feature.

**Comment by cenit** [comment]
yeah I was trying in fact to jot down a clear relationship, so that it does not get broken at the next revision. It's not so easy, also because there's no clear way to make features exclusive without "writing sentences too easy", as you'd say
